samza-core/src/main/java/org/apache/samza/table/remote/RemoteReadableTable.java

```diff
@@ -158,134 +174,30 @@ public V get(K key) {
       return CompletableFuture.completedFuture(Collections.EMPTY_MAP);
     }
     readMetrics.numGetAlls.inc();
-    return execute(readRateLimiter, keys, readFn::getAllAsync, readMetrics.getAllNs)
-        .handle((result, e) -> {
-            if (e != null) {
-              throw new SamzaException("Failed to get the records for " + keys, e);
-            }
-            return result;
-          });
-  }
-
-  /**
-   * Execute an async request given a table key
-   * @param key key of the table record
-   * @param method method to be executed
-   * @param timer latency metric to be updated
-   * @param <T> return type
-   * @return CompletableFuture of the operation
-   */
-  protected <T> CompletableFuture<T> execute(TableRateLimiter<K, V> rateLimiter,
```

Comment:
It seems that you have a common reuseable function here before. What's the reason to repeat it in many places now?